src/backend/catalog/objectaddress.c

```diff
@@ -978,9 +979,10 @@ get_object_address(ObjectType objtype, Node *object,
 													   &relation, missing_ok);
 				break;
 			case OBJECT_TRIGGER:
-				if(sql_dialect == SQL_DIALECT_TSQL)
-					address = get_object_address_trigger_tsql(castNode(List, object),
-													   	&relation, missing_ok);
+				if(get_trigger_object_address_hook){
```

Comment:
I would suggest adding a check for sql_dialect here instead of returning random values from the hook function if dialect is not tsql.

Reply:
Just saw Surendra's comment - I am fine with having it inside the hook function.